test/unit/query_context_cache_test.js

```diff
@@ -0,0 +1,220 @@
+const QueryContextCache = require('../../lib/queryContextCache.js');
+const assert = require('assert');
+
+function TestingQCC () {
+  this.qcc = null;
+  this.BASE_READ_TIMESTAMP = 1668727958;
+  this.CONTEXT = 'Some query Context';
+  this.BASE_ID = 0;
+  this.BASE_PRIORITY = 0;
+  this.MAX_CAPACITY = 5;
+  this.expectedIDs;
+  this.expectedReadTimestamp;
+  this.expectedPriority;
+  
+  this.initCache = function () {
+    this.qcc = new QueryContextCache(this.MAX_CAPACITY);
+  };
+  
+  this.initCacheWithData = function () {
+    this.initCacheWithDataWithContext(this.CONTEXT);
+  };
+  
+  this.initCacheWithDataWithContext = function (Context) {
+    this.qcc = new QueryContextCache(this.MAX_CAPACITY);
+    this.expectedIDs = [];
+    this.expectedReadTimestamp = [];
+    this.expectedPriority = [];
+    for (let i = 0; i < this.MAX_CAPACITY; i++) {
+      this.expectedIDs[i] = this.BASE_ID + i;
+      this.expectedReadTimestamp[i] = this.BASE_READ_TIMESTAMP + i;
+      this.expectedPriority[i] = this.BASE_PRIORITY + i;
+      this.qcc.merge(this.expectedIDs[i], this.expectedReadTimestamp[i], this.expectedPriority[i], Context);
+    }
+    this.qcc.syncPriorityMap();
+  };
+    
+  this.initCacheWithDataInRandomOrder = function () {
+    this.qcc = new QueryContextCache(this.MAX_CAPACITY);
+    this.expectedIDs = [];
+    this.expectedReadTimestamp = [];
+    this.expectedPriority = [];
+    for (let i = 0; i < this.MAX_CAPACITY; i++) {
+      this.expectedIDs[i] = this.BASE_ID + i;
+      this.expectedReadTimestamp[i] = this.BASE_READ_TIMESTAMP + i;
+      this.expectedPriority[i] = this.BASE_PRIORITY + i;
+    }
+  
+    this.qcc.merge(this.expectedIDs[3], this.expectedReadTimestamp[3], this.expectedPriority[3], this.CONTEXT);
+    this.qcc.merge(this.expectedIDs[2], this.expectedReadTimestamp[2], this.expectedPriority[2], this.CONTEXT);
+    this.qcc.merge(this.expectedIDs[4], this.expectedReadTimestamp[4], this.expectedPriority[4], this.CONTEXT);
+    this.qcc.merge(this.expectedIDs[0], this.expectedReadTimestamp[0], this.expectedPriority[0], this.CONTEXT);
+    this.qcc.merge(this.expectedIDs[1], this.expectedReadTimestamp[1], this.expectedPriority[1], this.CONTEXT);
+    this.qcc.syncPriorityMap();
+  };
+
+  this.assertCacheData = function () {
+    this.assertCacheDataWithContext(this.CONTEXT);
+  };
+  
+  this.assertCacheDataWithContext = function (Context) {
+    const size = this.qcc.getSize();
+    assert.strictEqual(size,this.MAX_CAPACITY);
+    const elements = Array.from(this.qcc.getElements());
+    for (let i = 0; i < size; i++) {
+      assert.strictEqual(this.expectedIDs[i], elements[i].getId());
+      assert.strictEqual(this.expectedReadTimestamp[i], elements[i].getReadTimestamp());
+      assert.strictEqual(this.expectedPriority[i], elements[i].getPriority());
+      assert.strictEqual(Context, elements[i].getContext());
+    }
+  };
+}
+
+describe('QueryContextCacheTest', function () {
```

Comment:
It is possible add some integration tests too? Test case about merged context is correctly send in next request is needed.

Reply:
I remember that during implementation of query context in GO driver @sfc-gh-pfus created tests I have in mind. Can you take a look to that scenarios: https://github.com/snowflakedb/gosnowflake/blob/082e9f808e6533c5110c12c0b2197c10d64b3e0e/htap_test.go

Reply:
I had a look, but I could not find an integration test that only tests the query context cache in that PR. If you do not mind, could you please tell me what the test name is?

Reply:
Like I've said previously I thought how to test that prepared query contexts are added to requests. I thinking about "TestHybridTablesE2E" but maybe you right - this test shows what we get from server not what we are sending. I'm thinking about control what is sent after adding queryContext to request for example in Function `sendSfRequest` but I've no idea if it is possible to control request like that. That's a pity but maybe we haven't possibility in current architecture :/

Reply:
I added the integration test in a new file testHTAP.js.